### ISSUE TITLE
Can't install angular-spinner via private-bower - "Arguments to path.resolve must be strings" error

### ISSUE THREAD

**Issue by HageMaster3108** [Hacklone/private-bower#79]
No sure how to re-open a issue, so I'm opening a new one:
Issue #71 has been closed as duplicate of #15 (which is now fixed through #78).
However the solution from #78 doesn't seem to fix this, hence I still think that this is a *different* issue.

when I try to install "angular-spinner" via a private-bower repository (using "bower install", bower.json see below), it dies with:
**"bower angular-spinner#0.5.1      error Arguments to path.resolve must be strings"**

My .bowerrc looks like this:
```
{
"registry": "http://localhost:5678"
}
```

This is a example bower.json which can be used to reproduce this
```
{
  "name": "Test",
  "version": "0.0.0",
  "dependencies": {
    "angular-spinner": "0.5.1"
  },
  "devDependencies": {}
}
```

My bower.conf.json is mostly default, the only things I changed are setting git/svn caches to enabled:
````
{
    "port": 5678,
    "registryFile": "./bowerRepository.json",
    "disablePublic": false,
    "publicRegistry": "http://bower.herokuapp.com/packages/",
    "authentication": {
        "enabled": false,
        "key": "password"
    },
    "repositoryCache": {
        "cachePrivate": false,
        "git": {
            "enabled": true,
            "cacheDirectory": "./gitRepoCache",
            "host": "localhost",
            "port": 6789,
            "publicAccessURL": null,
            "refreshTimeout": 10
        },
        "svn": {
            "enabled": true,
            "cacheDirectory": "./svnRepoCache",
            "host": "localhost",
            "port": 7891,
            "publicAccessURL": null,
            "refreshTimeout": 10
        }
    },
    "proxySettings" : {
        "enabled": false,
        "host": "proxy",
        "username": "name",
        "password" : "pass",
        "port": 8080,
        "tunnel": false
    },
    "log4js" : {
        "enabled": false,
        "configPath" : "log4js.conf.json"
    }
}
```

Any ideas?

I have debugged to the point where the error occurs in bower, it seems that it tries to resolve "undefined" as source, but I can't find out where the issue originates.

It seems that this happens after private-bower has newly mirrored a public package only.

So to reproduce this, you need to clear your private-bower gitRepoCache dir and bowerRepository.json.

Basically I can get the mirror to work if I run
```
rm -rf ~/.cache/bower; rm -rf bower_components; bower install
```
for each new dependency I have in my bower.json....

I guess there are two problems (assumptions):

* the response private-bower gives to bower once a public package has been freshly mirrored is somehow invalid
* that invalid response is mirrored by bower in its cache, so to retry you need to clear the bower local cache

**Comment by Hacklone** [comment]
I think it should be a problem with your local bower cache. Did you raise the version of your package?
If not then bower will cache that until you run: bower cache clean.

**Comment by Hacklone** [comment]
Closing this for inactivity.
